src/main/java/com/fundynamic/d2tm/game/math/BoundedVector2D.java

```diff
@@ -0,0 +1,33 @@
+package com.fundynamic.d2tm.game.math;
+
+
+public class BoundedVector2D<T extends Number> extends Vector2D {
```

Comment:
After working through the complete PR I now see how this is not a Rect. I think a better name is `ConstrainedVector2D` and it shouldn't derive from Vector2D; the two aren't interchangeable.